users/models.py

```diff
@@ -133,3 +133,27 @@ class Meta:
 
     def __str__(self):
         return f"UserCampaignRegistration {self.user.username}({self.key})"
+
+
+class UserActivityLog(models.Model):
+    """
+    This model stores user activity logs.
+    """
+
+    user = models.ForeignKey(User, null=True, on_delete=models.CASCADE)
+    ip_address = models.GenericIPAddressField(null=True, blank=True)
+    endpoint = models.CharField(max_length=200)
+    timestamp = models.DateTimeField(auto_now_add=True)
+    payload = models.JSONField(default=dict, blank=True, null=False)
+
+
+def log_user_activity(request):
+    user = request.user
+    if user.is_anonymous:
+        user = None
+    UserActivityLog.objects.create(
+        user=user,
+        ip_address=request.META.get("REMOTE_ADDR"),
```

Comment:
Does this work for requests being proxied? We have two proxies in front: nginx and CF, so remember to check if it will work in that case too